src/contexts/AuthContext.js

```diff
@@ -0,0 +1,31 @@
+import React, { createContext, useReducer } from "react";
+import AuthReducer from "reducers/AuthReducer";
+import API from "api/moviedb.instance";
+
+const AuthContext = createContext();
+
+export const AuthProvider = (props) => {
+    const [state, dispatch] = useReducer(AuthReducer, {
```

Comment:
I think you should use dispatch inside the `login`, `register`, and `logout` functions? Where you'll update the state and the local storage?